refinery/core/models.py

```diff
@@ -730,27 +716,15 @@ def get_file_size(self):
             [item.get_file_size(report_symlinks=True) for item in file_items]
         )
 
-    def get_isa_archive(self):
-        """Returns the isa_archive that was used to create the DataSet"""
-        investigation = self.get_investigation()
-        try:
-            return FileStoreItem.objects.get(uuid=investigation.isarchive_file)
-        except (FileStoreItem.DoesNotExist,
-                FileStoreItem.MultipleObjectsReturned,
-                AttributeError) as e:
-            logger.debug("Couldn't fetch FileStoreItem: %s", e)
-
-    def get_pre_isa_archive(self):
-        """Returns the pre_isa_archive that was used to create the DataSet"""
+    def get_metadata_as_file_store_item(self):
```

Comment:
This method appears to be unnecessary as this logic is repeated in the only place where it's used.

Reply:
I was referring to `if investigation.is_isatab_based` statement that is repeated both inside this method and in api.py which is a bit of a code smell.
As a matter of fact, there may not be a need for any of this logic if isa_archive and pre_isa_archive can be combined into a single field.